ray-curator/ray_curator/stages/io/reader/client_partitioning.py

```diff
@@ -0,0 +1,145 @@
+from dataclasses import dataclass, field
+from typing import Any, List
+import json
+import posixpath
+
+import fsspec
+from fsspec.core import url_to_fs
+
+from loguru import logger
+
+from ray_curator.backends.base import WorkerMetadata
+from ray_curator.stages.io.reader.file_partitioning import FilePartitioningStage
+from ray_curator.tasks import FileGroupTask, _EmptyTask
+from ray_curator.utils.storage_utils import get_files_relative, get_storage_client, read_json_file
+
+
+class FSPath:
+    """Wrapper that combines filesystem and path for convenient file operations."""
+    
+    def __init__(self, fs: fsspec.AbstractFileSystem, path: str):
+        self._fs = fs
+        self._path = path
+    
+    def open(self, mode: str = 'rb', **kwargs):
+        return self._fs.open(self._path, mode, **kwargs)
+    
+    def __str__(self):
+        return self._path
+    
+    def __repr__(self):
+        return f"FSPath({self._path})"
+
+
+@dataclass
+class ClientPartitioningStage(FilePartitioningStage):
+    """Stage that partitions input file paths from a client into FileGroupTasks.
+
+    This stage runs as a dedicated processing stage (not on the driver)
+    and creates file groups based on the partitioning strategy.
+    """
+
+    input_list_json_path: str | None = None
+    _name: str = "client_partitioning"
+    # internal
+    _fs: fsspec.AbstractFileSystem | None = field(default=None, init=False, repr=False)
+    _root: str | None = field(default=None, init=False, repr=False)
+
+    def setup(self, worker_metadata: WorkerMetadata | None = None) -> None:
+        self._fs, self._root = url_to_fs(self.file_paths, **(self.storage_options or {}))
```

Comment:
I think to align with earlier defaults we need to play with these storage options.  Main one is default_block_size, default_cache_type, default_fill_cache

```


storage_options = {
    "anon": False,                        # Set True for public S3
    "config_kwargs": {
        "max_pool_connections": 10,       # HTTP connection pool size
    },
    "default_block_size": 10 * 1024**2,     # --fs-block-size-mib (10 MiB)
    "default_cache_type": "readahead",     # --fs-cache-type (bytes | none | mmap | readahead)
    "default_fill_cache": True,            # Always fill cache for forward sequential reads
}
```

Reply:
CC: @ayushdg  for his input as well.